Formula/onedrive.rb

```diff
@@ -0,0 +1,25 @@
+class Onedrive < Formula
+  desc "Folder synchronization with OneDrive"
+  homepage "https://github.com/abraunegg/onedrive"
+  url "https://github.com/abraunegg/onedrive/archive/v2.4.5.tar.gz"
+  sha256 "1f1f5e1f2f37376b6d96bda2426552a94a8b195f545b4fb7f3668c4fe2e8f6a0"
+  license "GPL-3.0-only"
+
+  depends_on "dmd" => :build
+  depends_on "pkg-config" => :build
+  depends_on "curl-openssl"
+  depends_on :linux
+  depends_on "sqlite"
+
+  def install
+    ENV["DC"] = "dmd"
+    system "./configure", "--prefix=#{prefix}"
+    system "make", "install"
+    bash_completion.install "contrib/completions/complete.bash"
+    zsh_completion.install "contrib/completions/complete.zsh" => "_onedrive"
+  end
+
+  test do
+    system "#{bin}/onedrive", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.